internal/services/containers/kubernetes_cluster_scaling_resource_test.go

```diff
@@ -531,6 +589,44 @@ resource "azurerm_kubernetes_cluster" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, vmSize)
 }
 
+func (KubernetesClusterResource) updateOsDisk(data acceptance.TestData, osDiskType string, osDiskSize int) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  default_node_pool {
+    name                        = "default"
+    temporary_name_for_rotation = "temp"
+    node_count                  = 1
+    os_disk_type                = "%s"
+    os_disk_size_gb             = %d
```

Comment:
Should we also be changing the `os_sku` here to ensure that works?

Reply:
Done

Reply:
vm_size is missing
```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_updateOs
=== PAUSE TestAccKubernetesCluster_updateOs
=== CONT  TestAccKubernetesCluster_updateOs
testcase.go:110: Step 3/6 error: Error running pre-apply refresh: exit status 1
Error: Missing required argument
on terraform_plugin_test.tf line 28, in resource "azurerm_kubernetes_cluster" "test":
28:   default_node_pool {
The argument "vm_size" is required, but no definition was found.
testing_new.go:86: Error running post-test destroy, there may be dangling resources: exit status 1
Error: Missing required argument
on terraform_plugin_test.tf line 28, in resource "azurerm_kubernetes_cluster" "test":
28:   default_node_pool {
The argument "vm_size" is required, but no definition was found.
--- FAIL: TestAccKubernetesCluster_updateOs (251.45s)
FAIL
```

Reply:
Why are checks are green here? i would not expect that.

Reply:
The CI checks that are run on the PR don't run the acceptance tests - only unit tests, linting etc.

Reply:
good to know